src/mpid/ch4/shm/xpmem/xpmem_pre.h

```diff
@@ -43,13 +46,22 @@ typedef struct {
     MPIDI_XPMEM_segtree_t segcache;     /* AVL tree based segment cache */
 } MPIDI_XPMEM_segmap_t;
 
+typedef struct MPIDI_XPMEM_am_dmessage {
+    MPIDI_SHM_ctrl_hdr_t ctrl_hdr;
+    int ctrl_id;
+    struct MPIDI_XPMEM_am_dmessage *prev, *next;
+} MPIDI_XPMEM_am_dmessage_t;
+
 typedef struct {
     int num_local;
     int local_rank;
     MPIR_Group *node_group_ptr; /* cache node group, used at win_create. */
     xpmem_segid_t segid;        /* my local segid associated with entire address space */
     MPIDI_XPMEM_segmap_t *segmaps;      /* remote seg info for every local processes. */
+    OPA_int_t *xpmem_counter;
     size_t sys_page_sz;
```

Comment:
Can we use `coop_counters` instead of `xpmem_counter` ? `xpmem_` prefix here is meaningless.